ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/ForkAndMilestone.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.spec.logic;
```

Comment:
I'd probably move this to `tech.pegasys.teku.spec.datastructures.state` next to `Fork`?  Longer term, maybe it makes sense to move spec-defined "containers" to `tech.pegasys.teku.spec.containers` and leave everything else in `datastructures`?

Reply:
Went back and forth on those two options but settled on putting it in `...spec.containers` because it isn't part of the actual state so putting it in that package felt misleading.

Reply:
Is `spec.tuples` too weird 😂? Or `spec.datastructures.tuples`?  Would be nice to reserve the term "containers" for spec-related objects since it's used in the [spec](https://github.com/ethereum/eth2.0-specs/blob/0b069fbfcafb52201cf7f8a6e3ae0b0436931938/specs/phase0/beacon-chain.md#containers).

Reply:
I merged before I saw this but looking again it could actually go in `spec.datastructures.util` which is full of stuff that isn't from the actual spec (e.g. `AttestationProcessingResult`).  That feels reasonable to me - these are utility data structures we've created as opposed to something the spec defined.